openassessment/assessment/test/test_student_training_api.py

```diff
@@ -229,7 +229,7 @@ def test_assess_training_example_database_error(self):
             with self.assertRaises(StudentTrainingInternalError):
                 training_api.assess_training_example(self.submission_uuid, EXAMPLES[0]['options_selected'])
 
-    @ddt.data({}, {'num_required': 'not an integer!'})
+    @ddt.data({'some_other_field': 0}, {'num_required': 'not an integer!'})
```

Comment:
This test is now "`if requirements is not None or empty, but does not contain the field we expect` instead of `if requirements is not None`".

I'm not a huge fan of changing longstanding tests to pass when they fail this way, I'm a bit worried that maybe some other codepath depends on the old behavior. On the other hand, it seems super funky to have `student_training` error out on `{}`, while `peer` handles the same input just fine.